File: 5. Search In Linked List - GFG/5.-search-in-linked-list.cpp
// { Driver Code Starts
#include <bits/stdc++.h> 
using namespace std; 

struct Node
{
    int data;
    struct Node* next;
    
    Node(int x){
        data = x;
        next = NULL;
    }
};

bool searchLinkedList(struct Node* head, int x);

int main() 
{ 
    int t;
    cin>>t;
    while(t--)
    {
        int n;
        cin>>n;

        int data;
        cin>>data;
        struct Node *head = new Node(data);
        struct Node *tail = head;
        for (int i = 0; i < n-1; ++i)
        {
            cin>>data;
            tail->next = new Node(data);
            tail = tail->next;
        }
        int x; cin >> x;
        cout << searchLinkedList(head, x) << endl;
    }
    return 0;
}// } Driver Code Ends


//User function Template for C++


/*
struct Node
{
    int data;
    Node * next;
    Node (int x)
    {
        data=x;
        next=NULL;
    }
        
};
*/
bool searchLinkedList(Node *head, int x)
{
    //Your code here
    Node *curr = head;
    while(curr != NULL){
        if (curr->data == x){
            return 1;
        }
        curr = curr->next;
    }
    return 0;
}

File: Linked List/5. Search In Linked List - GFG/5.-search-in-linked-list.py
#User function Template for python3

'''
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
'''
def searchLinkedList(head,x):
    #code here
    curr = head
    while curr != None:
        if curr.data == x:
            return 1
        curr = curr.next
    return 0

#{ 
#  Driver Code Starts
#Initial Template for Python 3

# contributed by RavinderSinghPB
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


class Llist:
    def __init__(self):
        self.head = None

    def insert(self, data, tail):
        node = Node(data)

        if not self.head:
            self.head = node
            return node

        tail.next = node
        return node


if __name__ == '__main__':
    t = int(input())

    for tcs in range(t):

        n = int(input())
        arr = [int(x) for x in input().split()]
        x=int(input())

        ll = Llist()
        tail = None

        for nodeData in arr:
            tail = ll.insert(nodeData, tail)

        res=searchLinkedList(ll.head,x)
        print(res)
# } Driver Code Ends